src/scheduler/job_info.c

```diff
@@ -2255,6 +2263,14 @@ create_resresv_sets_resdef(status *policy, server_info *sinfo) {
 		defs[i++] = getallres(RES_PREEMPT_TARGETS);
 	defs[i] = NULL;
 
+	for(cur_res = limres; cur_res != NULL; cur_res = cur_res->next) {
```

Comment:
A minor style thing: your two for loops are missing spaces after the `for` keyword